projects/transformers/run_utils.py

```diff
@@ -1128,5 +1293,93 @@ def model_init():
                 writer.write(f"{key} = {value}\n")
 
 
-def compute_objective_eval_loss(metrics):
-    return metrics["eval_loss"]
+def compute_objective(metrics, objective):
+    return metrics[objective]
+
+
+def check_if_current_hp_best(old_file, model_args, best_run):
+    """
+    If you run multiple finetuning experiments, you want one file with
+    the best params / results. If the current run beats the old best,
+    overwrite the best_run_results file. This function returns a bool,
+    indicating if you should overwrite, or not.
+    """
+
+    # If there is no current fil, write a new file
+    if not os.path.exists(old_file):
+        return True
+
+    # Read the file with current best results
+    with open(old_file, "r") as f:
+        data = f.read()
+        line_split = data.split("\n")
+
+    # Search though the file to get the best score
+    previous_best = None
+    for line in line_split:
+        if model_args.hp_compute_objective[1] in line:
+            previous_best = float(line.split("=")[-1])
+
+    # To compare new scores with current scores, need to decide
+    # if greater is better.
+    if model_args.hp_compute_objective[0] == "maximize":
+        operator = "__gt__"
```

Comment:
Why not put `return previous_best > best_run.objective` here and `return previous_best < best_run.objective` in the else statement? This may be more readable.